First proposal how to add .well-known URI location to xRegistry (#125)
xregistry/spec: core/spec.md

 src="./xregfullmodel.png" height="300">&nbsp;&nbsp;&nbsp;<img
 src="./sample.png" height="300">

One purpose of the xRegistry is to enable easy and fully automated,
machine-readable discovery. The starting point of the discovery is via a
[well-known location](#registry-well-known-config): `/.well-known/xregistry.json`.

> This is relative to the base URL of the server.

Reviewer: "of the server". Does this mean of the host or of the xReg server? Meaning: http://example.com or http://example.com/xregistry  if the latter is the base URL of the xReg server?

Author: @duglin : In my understanding example.com would be the location to the server (and this would be its base url / hostname). /xregistry would be a path that actually is the entry point of an API that the server offers (but there may be other APIs as well).

Reviewer: Based on today's discussion I assume line 92 would change to allow for it to live some place else, right?

Author: > Based on today's discussion I assume line 92 would change to allow for it to live some place else, right?

I don't think so: The .well-known URI needs to be first discovered. But when you retrieve it, then you can point to the actual xRegistry server implementation(s) and their URLs. They could indeed live on a different server.

---

### Registry Well-Known Config

The registry comes with a well known location `/.well-known/xregistry.json`,
following [RFC8615](https://datatracker.ietf.org/doc/html/rfc8615).

It acts as a single entrypoint from where locations and capabilities of the
xRegistry implementation can be retrieved. Since this URL is always the same,
it can be used to find out whether xRegistry is supported and where to find the API.
It also allows to support multiple versions of the xRegistry protocol at the same time
and allows implementers to choose their own API base path.

> The capabilities described on this level are independent from the xRegistry protocol version or its current meta model.

Author: @duglin : I've updated the PR a little bit to reflect our discussions today.

---

### Registry Well-Known Config

The registry comes with a well known location `/.well-known/xregistry.json`,
following [RFC8615](https://datatracker.ietf.org/doc/html/rfc8615).

It acts as a single entrypoint from where locations and capabilities of the
xRegistry implementation can be retrieved. Since this URL is always the same,
it can be used to find out whether xRegistry is supported and where to find the API.

> It also allows to support multiple versions of the xRegistry protocol at the same time

Reviewer: ...allows to support multiple versions...  ->  ..allows for support of multiple versions...

Author: ```suggestion
It allows for support of multiple versions of the xRegistry protocol at the same time
```

and allows implementers to choose their own API base path.

The capabilities described on this level are independent from the xRegistry protocol version or its current meta model.
They resemble the capabilities of the xRegistry implementation itself that consumers can rely upon.

```yaml
{
  # Some high-level description of the xRegistry that are independent from version of the protocol nor its meta model
  "description": "STRING", ?
  "documentation": "URL", ?
  "labels": { "STRING": "STRING" * }, ? #

  # Describe where to find the xRegistry API and its capabilities
  "apis": [{ # Major version of the xRegistry API protocol. In the future, there could be v2 in parallel.
    "specversion": "STRING", # full specversion, as returned by the API root response
    "apiurl": "URL", ? # Point to root URL of xRegistry API, e.g. /x-registry/v1/
    "capabilities": ["write", "update", "inline", "filter", "..."] # TODO: TBD

>     "modelurl": "URL", ? # Point to registry meta model

Reviewer: "modelurl" seems like it would be a duplicate of the `/model` API that's part of the API itself. no?
Or do you see cases where there's no "apiurl" but still a "modelurl" ?
I guess it depends on whether the "files" stuff that we've talked about includes the model definition or whether it's just the metadata.

Author: yes. I actually wasn't really sure if we should include it or not. I put it in, so it can be discussed.

---

### Registry Well-Known Config

The registry comes with a well known location `/.well-known/xregistry.json`,
following [RFC8615](https://datatracker.ietf.org/doc/html/rfc8615).

It acts as a single entrypoint from where locations and capabilities of the
xRegistry implementation can be retrieved. Since this URL is always the same,
it can be used to find out whether xRegistry is supported and where to find the API.
It also allows to support multiple versions of the xRegistry protocol at the same time
and allows implementers to choose their own API base path.

```yaml
{
  # The following top-level attributes are identical to the xRegistry API root response

>   # We may consider removing them here if we're bothered by the duplication.

Reviewer: Some of these (like labels) might be container private data, and "self" wouldn't be accurate. It might be best to either remove these or sub-set them down to just simple (safe) things like "id", and make it optional since even that could expose too much info.

  "id": "STRING",
  "name": "STRING", ?
  "epoch": UINTEGER,
  "self": "URL",
  "description": "STRING", ?
  "documentation": "URL", ?
  "labels": { "STRING": "STRING" * }, ?
  "createdat": "TIME",
  "modifiedat": "TIME",

  # Describe where to find the xRegistry API and its capabilities

>   "apis": {

Author: Good point. First I wanted to make this an array, but then I remembered that you often preferred to have dictionaries. But since you don't need to have links to this, I would then propose something like:

```ts
apis: Array<{ version: "v1" | "v2", url: "URL" }>
```
